File: .sdk/tm/php/utility/ResultBody.php
<?php
declare(strict_types=1);

// Thesmsworks SDK utility: result_body

class ThesmsworksResultBody
{
    public static function call(ThesmsworksContext $ctx): ?ThesmsworksResult
    {
        $response = $ctx->response;
        $result = $ctx->result;
        if ($result && $response && $response->json_func && $response->body) {
            $result->body = ($response->json_func)();
        }
        return $result;
    }
}

File: java/utility/ResultBody.java
package voxgig.thesmsworkssdk.utility;

import voxgig.thesmsworkssdk.core.Context;
import voxgig.thesmsworkssdk.core.Response;
import voxgig.thesmsworkssdk.core.Result;

final class ResultBody {

  private ResultBody() {}

  static Result resultBody(Context ctx) {
    Response response = ctx.response;
    Result result = ctx.result;

    if (result != null) {
      if (response != null && response.jsonFunc != null && response.body != null) {
        result.body = response.jsonFunc.get();
      }
    }

    return result;
  }
}
